src/Features/Core/Portable/InitializeParameter/AbstractInitializeParameterCodeRefactoringProvider.cs

```diff
@@ -28,62 +29,131 @@ internal abstract partial class AbstractInitializeParameterCodeRefactoringProvid
+                if (parameter == null || parameter.Name == "")
+                {
+                    return;
+                }
+
+                var methodSymbol = (IMethodSymbol)parameter.ContainingSymbol;  // foo(objact a,
+                if (methodSymbol.IsAbstract ||
+                    methodSymbol.IsExtern ||
+                    methodSymbol.PartialImplementationPart != null ||
+                    methodSymbol.ContainingType.TypeKind == TypeKind.Interface)
+                {
+                    continue;
+                }
+
+                //var blockStatementOpt = GetBlockStatmentOpt(syntaxFacts, semanticModel, functionDeclaration, cancellationToken);
+                //if (blockStatementOpt == null)
+                //{
+                //    continue;
+                //}
+                if (CanOfferRefactoring(functionDeclaration, semanticModel, syntaxFacts, cancellationToken, out var blockStatementOpt))
+                {
+                    // Ok.  Looks like there is at least one reasonable parameter to analyze.  Defer to subclass to 
+                    // actually determine if there are any viable refactorings here.
+
+                    // Update the list of valid parameter nodes
+                    listOfPotentiallyValidParametersNodes.Add(parameterNode);
+
+                    // For single parameter - Only offers for the parameter cursor is on
+                    if (firstparameterNode == parameterNode)
+                    {
+                        context.RegisterRefactorings(await GetRefactoringsForSingleParameterAsync(
+                    document, parameter, functionDeclaration, methodSymbol, blockStatementOpt, cancellationToken).ConfigureAwait(false));
+                    }
+
+                    // Calls for a multiple null check only when this is the last iteration and there's more than one possible valid parameter parameter
+                    if (listOfPotentiallyValidParametersNodes.Count > 1 && counter == parameterNodes.Count())
+                    {
+                        context.RegisterRefactorings(await GetRefactoringsForAllParametersAsync(
+                            document, functionDeclaration, methodSymbol, blockStatementOpt, listOfPotentiallyValidParametersNodes, position, cancellationToken).ConfigureAwait(false));
+                    }
+                }
             }
         }
 
-        private bool CanOfferRefactoring(SyntaxNode functionDeclaration, SemanticModel semanticModel, ISyntaxFactsService syntaxFacts, CancellationToken cancellationToken, out IBlockOperation blockStatementOpt)
+        protected virtual Task<ImmutableArray<CodeAction>> GetRefactoringsForAllParametersAsync(Document document, SyntaxNode functionDeclaration, IMethodSymbol method, IBlockOperation blockStatementOpt, IEnumerable<SyntaxNode> listOfParameterNodes, int position, CancellationToken cancellationToken)
+        {
+            return Task.FromResult(ImmutableArray<CodeAction>.Empty);
```

Comment:
why?

Reply:
What do you suggest should go there instead?

Reply:
well, should this not just be abstract, if we need both types to implement this?

Reply:
agreed! thanks.